src/Searchlight/DataSource.cs

```diff
@@ -621,6 +621,17 @@ private static void Expect(string expectedToken, string actual, string originalF
             }
         }
 
+        private static object DefinedDateOperators(string valueToken)
+        {
+            if (StringConstants.DefinedDates.Keys.Contains(valueToken, StringComparer.OrdinalIgnoreCase))
```

Comment:
Hmm.  I think this evaluates all keys linearly rather than making use of the dictionary's hash table.

Since there's only three items in the dictionary now, it's not too big of a deal though.